### PR TITLE
feat: AWS Lambda Powertools for Java templates

### PR DESCRIPTION
**Pull request by am29d**
*Issue #, if available:* #337 

*Description of changes:*

Creating three new templates for runtimes java8, java8.al2 and java11, for maven and gradle build tool respectively. These include allowing customers to selectively add AWS Lambda Powertools Tracing, Logging and Metrics utilities.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by am29d** [comment]
I checked the error logs from the last CodeBuild run and there seems to be some issue with Gradle:
> Stdout: Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details

I added integration tests for the folders, and if I run the same command as the CI they are green. Is there a specific configuration on the CI server that I need to replicate on my local machine?

**Review by jfuss — CHANGES_REQUESTED** [review]
Some small requests. I assume these were validated in some way (probably tests?), because it's hard to follow with all the cookiecutter ifs and such.

File: nodejs14.x/hello-ts-pt/cookiecutter.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "architectures": {
         "value": []
     },
-    "Powertools X-Ray Tracing": ["enabled","disabled"],
+    "Powertools Tracing": ["enabled","disabled"],
```
[the remark this answers is not in the document]
Comment (am29d) [reply nodejs14.x/hello-ts-pt/cookiecutter.json:7]: This was an accident, will scope it only for Java.

**Comment by am29d** [comment]
> Some small requests. I assume these were validated in some way (probably tests?), because it's hard to follow with all the cookiecutter ifs and such.

I have validated all 6 templates manually. I have added the unit tests, similar what we already have for other projects.

File: tests/integration/unit_test/test_unit_test_java11.py [hunk 2]
```diff
@@ -12,6 +12,14 @@ class UnitTest_java11_cookiecutter_aws_sam_hello_java_maven(UnitTestBase.JavaUni
     directory = "java11/hello-maven"
     code_directories = ["HelloWorldFunction"]
 
+class UnitTest_java11_cookiecutter_aws_sam_hello_java_powertools_maven(UnitTestBase.JavaUnitTestMavenBase):
+    directory = "java11/hello-powertools-maven"
+    code_directories = ["HelloWorldFunction"]
+
+class UnitTest_java11_cookiecutter_aws_sam_hello_java_powertools_gradle(UnitTestBase.JavaUnitTestGradleBase):
+    directory = "java11/hello-powertools-gradle"
```
[the remark this answers is not in the document]
Comment (am29d) [reply tests/integration/unit_test/test_unit_test_java11.py:20]: Completely forgot about the tests, sorry 🤦‍♂️

**Comment by lucashuy** [comment]
Was not able to reproduce the Java 11 unit test failures by running the test on my local system, I'll merge from master to kick off the codebuild again to see if this is still a problem.

**Comment by heitorlessa** [comment]
@am29d could you remove Gradle from this template so we can merge it? I'll remove from the docs too

**Comment by am29d** [comment]
I see you opened the PR without gradle, so I am closing this PR
